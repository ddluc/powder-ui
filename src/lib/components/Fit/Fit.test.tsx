import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Fit, FitProps } from './index';

const args: FitProps = {
  align: '50% 50%',
  fit: 'cover'
};

// Define unit tests
describe('Fit', () => {
  afterEach(cleanup);

  it('should render with defined props', () => {
    const result = render(
      <Fit {...args}>
        <img src="#" alt="A fun image" />;
      </Fit>
    );
    const component = result.container.firstChild;
    expect(component).toMatchSnapshot();
  });
});
