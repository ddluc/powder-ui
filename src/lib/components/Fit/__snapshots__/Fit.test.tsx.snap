// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fit should render with defined props 1`] = `
.c0 {
  display: block;
  height: auto;
  width: auto;
  position: relative;
}

.c1 > * {
  object-fit: cover;
  height: 100%;
  width: 100%;
  object-position: 50% 50%;
}

<div
  class="c0 c1"
>
  <img
    alt="A fun image"
    src="#"
  />
  ;
</div>
`;
